src/pipelines/rust_app.rs

```diff
@@ -241,6 +236,39 @@ impl RustApp {
             wasm_output: hashed_wasm_name,
         })
     }
+
+    async fn wasm_opt_build(&self, hashed_name: &str) -> Result<()> {
+        // Zero means no optimizations (valid arg for wasm-opt), so we can skip calling wasm-opt as
+        // it wouldn't have any effect.
+        if self.wasm_opt.as_deref() == Some("0") {
+            return Ok(());
+        }
```

Comment:
I would say that we should do:
```rust
let wasm_opt = match self.wasm_opt.as_deref().as_str() {
    Some(val) if val != "0" => val,
    _ => return Ok(());
};
```

Basically this will allow Trunk workflows to continue to work even without wasm-opt present on their machine. If they decide that they want to start using it, then they can set it to a non-zero value, and they are g2g.

Once we implement the work to actually download wasm-opt for the user (and probably download wasm-bindgen-cli for them as well), then we can make this default to opt level `4` or something like that, as it will be mostly transparent to the user and will not break existing workflows.

Thoughts?

Reply:
Yeah that's probably a better approach and would not cause half the users to get errors when running Trunk after the update 😅.

I think `""` (no opt level) should be the default as that already applies many optimizations (maybe even equivalent to `3` or `4`) and `wasm-pack` does the same.